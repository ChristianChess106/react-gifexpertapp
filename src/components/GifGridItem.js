import React from 'react'

export const GifGridItem = ({id,title,url}) => {

    /* console.log(id,title); */
    return (
        <div className="card animate__fadeInDown">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
