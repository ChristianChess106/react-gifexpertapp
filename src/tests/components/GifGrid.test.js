import React from "react";
import "@testing-library/jest-dom";
import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('pruebas con gifgrid', () => {

    const category = "Goku";
   
    //let wrapper = shallow(<GifGrid category={category}/>);
    test('testeando el componente ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

     test('mostrar items cuando se cargan imagenes useFetchGifs ', () => {

        const gifs = [{
            id: "ABC",
            title: "el ara√±a",
            url: "https://localhost.jpg"
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
       const wrapper = shallow(<GifGrid category={category}/>);

       expect(wrapper.find("p").exists()).toBe(false);
       expect(wrapper.find("GifGridItem").length).toBe(gifs.length);

      
        
    }); 
    
    
})
