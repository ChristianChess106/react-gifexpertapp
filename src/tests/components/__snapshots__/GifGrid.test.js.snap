// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con gifgrid testeando el componente  1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
